scripts/pcm-dpc/covid-19/preprocess.py

```diff
@@ -0,0 +1,180 @@
+# Lint as: python3
+"""
+This script preprocesses the covid19 data from Italy Department of Civil
+Protection for importing into Data Commons.
+"""
+
+import pandas as pd
+
+class PcmDpc:
+    """download the csv files and preprocess it for importing into
+    DataCommons"""
+    def preprocess(self):
+        """clean and save the CSV file"""
+        self.data = pd.read_csv(self.csvpath).drop(
+            columns=['note'])
+        self._translate()
+        self.data['Date'] = self.data['Date'].str[0:10]
+        self.set_location() # prepocess the geo ids
+        self.data.to_csv(self.name + '.csv', index=False)
+
+    def generate_tmcf(self):
+        """ generate the template mcf"""
+        geo_node = self.geo_template() # write the geo node to template mcf
+        TEMPLATE = ('Node: E:pcm-dpc->E{index}\n'
+                    'typeOf: dcs:StatVarObservation\n'
+                    'variableMeasured: dcs:{SVFullName}\n'
+                    'observationAbout: {geoNode}\n'
+                    'observationDate: C:pcm-dpc->Date\n'
+                    'value: C:pcm-dpc->{SVname}\n\n')
+        idx = 1
+        with open(self.name + '.tmcf', 'a') as f_out:
+            for statvar in self.stat_vars:
+                if statvar == 'CumulativeTestsPerformed':
+                    sv_full_name = 'cumulativeCount_MedicalTest_COVID_19'
+                elif statvar == 'CumulativeTestedPeople':
+                    sv_full_name = 'cumulativeCount_Person_COVID_19_Tested'
+                elif 'Cumulative' in statvar:
+                    sv_full_name = ('cumulativeCount_MedicalConditionIncident_'
+                                    'COVID_19_'+ statvar.replace('Cumulative', ''))
+                elif 'Incremental' in statvar:
+                    sv_full_name = ('incrementalCount_MedicalConditionIncident_'
+                                    'COVID_19_' + statvar)
+                else:
+                    sv_full_name = ('count_MedicalConditionIncident_COVID_19_'
+                                    + statvar)
+                f_out.write(TEMPLATE.format_map({'index': idx,
+                    'SVname': statvar, 'SVFullName': sv_full_name,
+                    'geoNode': geo_node}))
```

Comment:
Curious if this is better than just having a frozenset of StatVar ids?

Reply:
If we had just translated the columns of the cleaned CSV to be the StatVar ID, we would be able to reduce a lot of code, right?

Reply:
Agree.